storage/namespace.go

```diff
@@ -149,15 +151,24 @@ func newDatabaseNamespace(
 	increasingIndex increasingIndex,
 	writeCommitLogFn writeCommitLogFn,
 	opts Options,
-) databaseNamespace {
-	id := metadata.ID()
+) (databaseNamespace, error) {
+	if err := metadata.Validate(); err != nil {
```

Comment:
Once the registry is kv backed, this ctor could be called by a `database` which has already been running so we'll need the check.